### PR TITLE
Support Pair default fee in RouterV2

### PR DESCRIPTION
**Pull request by JanKuczma**

### CODE REVIEW

File: amm/contracts/factory/Cargo.toml [hunk 1]
```diff
@@ -7,10 +7,10 @@ edition = "2021"
 [dependencies]
 ink = { version = "=4.3.0", default-features = false }
 
-scale = { package = "parity-scale-codec", version = "3", default-features = false, features = [
+scale = { package = "parity-scale-codec", version = "=3.6.9", default-features = false, features = [
     "derive",
 ] }
-scale-info = { version = "2.9", default-features = false, features = [
+scale-info = { version = "=2.10.0", default-features = false, features = [
```
Comment (ggawryal) [inline amm/contracts/factory/Cargo.toml:13]: Why is fixing all those versions needed?

**Review by ggawryal — APPROVED** [review]
LGTM

File: amm/contracts/router_v2/pair.rs [hunk 2]
```diff
@@ -28,15 +30,23 @@ impl Pair {
...
         let contract_ref: contract_ref!(PairTrait, Env) = pair_id.into();
-        // Assume that the `pair_id` is a Pair contract and try to get the fee value.
+
+        // Assume that the `pair_id` is a Pair contract and try to get the token_0 address.
         // If the call is not successful return None indicating that the `pair_id`
-        // is not a Pair contract
-        let fee = match contract_ref.call().get_fee().try_invoke() {
-            Ok(Ok(fee)) => fee,
+        // is not a Pair contract.
+        let token_0 = match contract_ref.call().get_token_0().try_invoke() {
```
Comment (woocash2) [inline amm/contracts/router_v2/pair.rs:37]: Why suddenly like this?

**Review by woocash2 — APPROVED** [review]
Thank you sir

File: amm/contracts/factory/Cargo.toml (hunk 1 above)
Comment (JanKuczma) [reply amm/contracts/factory/Cargo.toml:13]: The rust version for this project is fixed to `1.72.1`. `scale-info` has `derive_more` as a dependency, which in `scale-info` > `2.11.3` requires rust `1.75.0` or newer.

**Pull request closed without merging**